### ISSUE TITLE
gitea.event_name set to push when running scheduled

### ISSUE THREAD

**Issue by nephatrine** [go-gitea/gitea#27667]
### Description

I have a workflow that runs both on push and on a cron schedule. When I echo `${{ gitea.event_name }}` though it shows `push` both when I push it out and when it runs due to the schedule. From what I can tell, GitHub sets the `event_name` to `schedule` when running from a schedule so I expected Gitea to work similarly.

Is this a bug? If not, is there an alternate means to have a workflow detect which trigger it is running under?

### Gitea Version

1.21.0-rc1

### Can you reproduce the bug on the Gitea demo site?

No

### Git Version

2.40

### Operating System

alpine:latest

### How are you running Gitea?

Gitea self-built and running on a docker image. Recently "updated" from 1.20.5 to 1.21.0-rc1 to for the cron actions.

### Database

SQLite

**Comment by lstellway** [comment]
I see this as well. 

`gitea v1.21.0-rc1`
`gitea-runner v0.2.6`

Cross-referencing the implementation PR:
https://github.com/go-gitea/gitea/pull/26655

Not entirely sure, but it looks like a `schedule` value may need to be added here:

- [modules/webhook/type.go#L11-L33](https://github.com/go-gitea/gitea/blob/ec0c6829d4c1096bfa0f9feb52b87b70fba014ea/modules/webhook/type.go#L11-L33)
- [modules/webhook/type.go#L37C9-L68](https://github.com/go-gitea/gitea/blob/ec0c6829d4c1096bfa0f9feb52b87b70fba014ea/modules/webhook/type.go#L37C9-L68)

So it is returned here:

- [routers/api/actions/runner/utils.go#L124-L126](https://github.com/go-gitea/gitea/blob/ec0c6829d4c1096bfa0f9feb52b87b70fba014ea/routers/api/actions/runner/utils.go#L124-L126)

**Comment by fuxiaohei** [comment]
I test main branch with yaml:

```yaml
on:
  schedule:
    - cron: "*/5 * * * *"

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - name: Make some test files
        run: |
          echo "🎉 The job was automatically triggered by a ${{ github.event_name }} event."
          echo "🎉 The job was automatically triggered by a ${{ gitea.event_name }} event."
```

Action prints:
<img width="608" alt="image" src="https://github.com/go-gitea/gitea/assets/2142787/82802964-8391-4264-933b-182620860ff1">

**Comment by lunny** [comment]
So maybe it's a 1.21-only problem.

**Comment by t4skforce** [comment]
I see this as well.

gitea v1.21.1
act_runner v0.2.6

```
on:
    push:
    schedule:
        - cron: "*/5 * * * *"

jobs:
    build:
        runs-on: ubuntu-latest

        steps:
            - name: Make some test files
              run: |
                echo "🎉 The job was automatically triggered by a ${{ github.event_name }} event."
                echo "🎉 The job was automatically triggered by a ${{ gitea.event_name }} event."
```

![image](https://github.com/go-gitea/gitea/assets/7422037/0182123d-b7c8-464c-b93d-e23badd4c137)

Build via "push"
![image](https://github.com/go-gitea/gitea/assets/7422037/92f3bc03-ef9f-4944-ad31-07fff87c2483)

Build via "schedule"
![image](https://github.com/go-gitea/gitea/assets/7422037/6abbf9f1-8b1e-4510-baae-ef941ab9a5ea)

**Comment by henrygoodman** [comment]
> I see this as well.
> 
> gitea v1.21.1 act_runner v0.2.6
> 
> ```
> on:
>     push:
>     schedule:
>         - cron: "*/5 * * * *"
> 
> jobs:
>     build:
>         runs-on: ubuntu-latest
> 
>         steps:
>             - name: Make some test files
>               run: |
>                 echo "🎉 The job was automatically triggered by a ${{ github.event_name }} event."
>                 echo "🎉 The job was automatically triggered by a ${{ gitea.event_name }} event."
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/7422037/285757332-0182123d-b7c8-464c-b93d-e23badd4c137.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExNjY3MTMsIm5iZiI6MTcwMTE2NjQxMywicGF0aCI6Ii83NDIyMDM3LzI4NTc1NzMzMi0wMTgyMTIzZC1iN2M4LTQ2NGMtYjkzZC1lMjNiYWRkNGMxMzcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTEyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzExMjhUMTAxMzMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTZlOGY0NWZjMGEyMzViZWY4NzlmZDA2YWZlMmYxNjhiMWQxN2U3ZjhmNjZlMDFiNjliYzA5NGE5YTVkYzc2NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.g4JPdUabl7nPtR-SsWgjVJZdqD8lnhbzABo7K7-QEJ4)
> 
> Build via "push" ![image](https://private-user-images.githubusercontent.com/7422037/285757471-92f3bc03-ef9f-4944-ad31-07fff87c2483.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExNjY3MTMsIm5iZiI6MTcwMTE2NjQxMywicGF0aCI6Ii83NDIyMDM3LzI4NTc1NzQ3MS05MmYzYmMwMy1lZjlmLTQ5NDQtYWQzMS0wN2ZmZjg3YzI0ODMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTEyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzExMjhUMTAxMzMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWQ2OTRiNDk3Zjg2YWI0NDg3YTk5NDE1OTdhZjdiNWZhZjk3NGNlNGQwYzMwYTRmNjAyZGE0M2Y4MWJkMDQ0NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.TYsrPOK6WlmcvGylzf6bT6k1CqzcMC05yzc1WfNP5lk)
> 
> Build via "schedule" ![image](https://private-user-images.githubusercontent.com/7422037/285757646-6abbf9f1-8b1e-4510-baae-ef941ab9a5ea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExNjY3MTMsIm5iZiI6MTcwMTE2NjQxMywicGF0aCI6Ii83NDIyMDM3LzI4NTc1NzY0Ni02YWJiZjlmMS04YjFlLTQ1MTAtYmFhZS1lZjk0MWFiOWE1ZWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTEyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzExMjhUMTAxMzMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTIxZWVlMmU3M2YyM2M5ZDhiZDVmYTcxN2RlZDFhMmNiMzc2ODJlYjE1ZTIxOGJkNmMxYTAxMzM4ZDA5YWQxZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.NgSMWID-1n9SwxJrq513CNG33ulzPxD82ofWFCCk8dQ)
> 
> It seems the issue is with having two triggers, one on push and one on schedule.

In the `Set up job` step, what does it say for the `this job was triggered by event: ...` (paraphrasing)

**Comment by t4skforce** [comment]
Build via "push":
```
act-org-github(version:v0.2.6) received task 713 of job 708, be triggered by event: push
workflow prepared
evaluating expression 'success()'
expression 'success()' evaluated to 'true'
```

Build via "schedule": 
```
act-org-github(version:v0.2.6) received task 714 of job 709, be triggered by event: push
workflow prepared
evaluating expression 'success()'
expression 'success()' evaluated to 'true'
```

**Comment by phadric** [comment]
I have the same Problem.
It actually goes a bit further.

For testing I created a new Repo with the following workflow:


.github/workflows/test.yml:
```
on:
    push:
    schedule:
        - cron: "*/5 * * * *"

jobs:
    build:
        runs-on: ubuntu-latest

        steps:
            - name: Make some test files
              run: |
                echo "🎉 The job was automatically triggered by a ${{ github.event_name }} event."
                echo "🎉 The job was automatically triggered by a ${{ gitea.event_name }} event."
```

It never gets executed.
Then I create a second workflow:
.github/workflows/push.yml
```
on:
  push:

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - name: Make some test files
        run: |
          echo "🎉 The job was automatically triggered by a ${{ github.event_name }} event."
          echo "🎉 The job was automatically triggered by a ${{ gitea.event_name }} event."   
```

After this the first workflow is executed every 2 minutes.
I checked my other repos with scheduled Jobs. They seem to use whatever event happened last for github.event_name regardless if they triggered a Workflow or not.

**Comment by Zettat123** [comment]
Close this issue since the bug has been fixed by #28691
